packages/nvidia_nat_mcp/src/nat/plugins/mcp/auth/service_account/provider.py

```
@@ -0,0 +1,136 @@
+    """
+    MCP service account authentication provider using OAuth2 client credentials.
+
+    Provides headless authentication for MCP clients using service account credentials.
+    Supports two authentication patterns:
+
+    1. Single authentication: OAuth2 service account token only
+    2. Dual authentication: OAuth2 service account token + service-specific token
+
+    """
+
+    def __init__(self, config: MCPServiceAccountProviderConfig, builder=None):
```

Ruff (0.14.4)

Unused method argument: `builder`

(ARG002)